src/components/ExtractorResponseEditor/ExtractorResponseEditor.tsx

```diff
@@ -422,16 +422,40 @@ class ExtractorResponseEditor extends React.Component<Props, State> {
         document.removeEventListener("Test_SelectWord", this.onTestSelectWord)
     }
 
-    // For end 2 end unit testing.
+    /**
+     * For cypress end to end unit testing.  Event selects phrase in entity labeller
+     * The following will select the word "hello" in the second row (row is 1 based)
+     *   var event = new CustomEvent("Test_SelectWord", { detail: { phrase: "hello", r:ow: 2 }})
+     *   event.initEvent("Test_SelectWord", true, true)}
+     *   element.dispatchEvent(event)
+     * 
+     * Or default to the first row:
+     *   var event = new CustomEvent("Test_SelectWord", { detail: "hello"})
+     *   event.initEvent("Test_SelectWord", true, true)}
+     *   element.dispatchEvent(event)
+     */
     @autobind
     onTestSelectWord(val: any) {
-        const phrase: string = val.detail
+        if (!val.detail) {
+            throw new Error("Test_SelectWord expecting detail phrase")
+        }
+
+        const phrase: string = val.detail.phrase || val.detail
+        const row: number = val.detail.row - 1 || 0
```

Comment:
I don't think the ` || 0` is correct.  I think the `- 1` would coerce the first portion into a number regardless of `row` value so ` || 0` would never evaluate an you could still get negative numbers

I believe it would be better to use `Math.max(val.detail.row - 1, 0)`

Reply:
good catch.  Missed then when I updated it to being 1 based

Reply:
Should be:
    const row: number = val.detail.row ? (val.detail.row - 1) : 0

Reply:
Don't understand why this was marked as resolved while not using the `max` which is safer.

It seems `{ detail: '', row: -4 }` is valid in your checks. Meaning -4 - 1 is -5
Which would attempt to do `rows[-5]` instead of fallback to 0 as `row[0]`

Reply:
The goad here isn't to validate user input, it's to differentiate between the two data types